exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/ColumnType.kt

```diff
@@ -908,6 +909,53 @@ class EnumerationNameColumnType<T : Enum<T>>(
     }
 }
 
+// JSON columns
+
+/**
+ * Column for storing JSON data, either in non-binary text format or the vendor's default JSON type format.
+ */
+open class JsonColumnType<T : Any>(
+    /** Returns the function that encodes an object of type [T] to a JSON String. */
+    val serialize: (T) -> String,
+    /** Returns the function that decodes a JSON String to an object of type [T]. */
+    val deserialize: (String) -> T
+): ColumnType() {
+    override fun sqlType(): String = currentDialect.dataTypeProvider.jsonType()
+
+    override fun valueFromDB(value: Any): Any {
+        return when (value) {
+            is PGobject -> deserialize(value.value!!)
+            else -> deserialize(value as String)
+        }
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    override fun notNullValueToDB(value: Any) = serialize(value as T)
+
+    override fun setParameter(stmt: PreparedStatementApi, index: Int, value: Any?) {
+        val parameterValue = when (currentDialect) {
+            is PostgreSQLDialect -> PGobject().apply {
+                type = sqlType()
+                this.value = value as String?
+            }
+            else -> value
+        }
+        super.setParameter(stmt, index, parameterValue)
+    }
+}
+
+/**
+ * Column for storing JSON data in binary format.
+ */
+class JsonBColumnType<T : Any>(
+    /** Returns the function that encodes an object of type [T] to a JSON String. */
```

Comment:
We can skip KDoc for parameters or write it as `@param` in the constructor or class KDoc